trio/_channel.py

```diff
@@ -206,14 +235,19 @@ def clone(self):
             raise trio.ClosedResourceError
         return MemorySendChannel._create(self._state)
 
-    def __enter__(self):
+    def __enter__(self) -> "MemorySendChannel[SendType]":
```

Comment:
This should return `Self`. (from `typing_extensions`, that is). (unlike `clone` above, this doesn't always return `MemorySendChannel`.)

If we don't want to use that, we can do `def __enter__(self: SelfT) -> SelfT` where `SelfT = TypeVar("SelfT")`.